packages/core/src/overlay/OverlayPanelCloseButton.css

```diff
@@ -4,12 +4,12 @@
   z-index: 1;
 }
 
-.saltOverlayPanelCloseButton-button.saltButton-secondary {
-  position: sticky;
-  right: 0;
-  top: 0;
-}
-
-.saltOverlayPanelCloseButton ~ .saltOverlayPanelContent {
-  padding-right: var(--salt-spacing-400);
+.saltOverlayPanel-inner:has(.saltOverlayPanelCloseButton) {
```

Comment:
`:has` isn't available in [supported browser](https://www.saltdesignsystem.com/salt/about/supported-platforms)?

Reply:
i don't know whether form field's grid slot approach would be easier?

Reply:
Oh wow - didn't realise we went back that far! 😱 Will re-visit with a grid option 👍

Reply:
Switched to grid approach that works:
* If close button comes before panel content in source order
* If close button comes after panel content
* If close button is not present